backend/component/sheet/sheet.go

```diff
@@ -0,0 +1,47 @@
+package sheet
+
+import (
+	"context"
+	"log/slog"
+	"strings"
+
+	"github.com/bytebase/bytebase/backend/plugin/parser/base"
+	"github.com/bytebase/bytebase/backend/store"
+	storepb "github.com/bytebase/bytebase/proto/generated-go/store"
+)
+
+func CreateSheet(ctx context.Context, s *store.Store, sheet *store.SheetMessage) (*store.SheetMessage, error) {
+	if sheet.Payload == nil {
+		sheet.Payload = &storepb.SheetPayload{}
+	}
+	sheet.Payload.Commands = getSheetCommands(sheet.Payload.Engine, sheet.Statement)
+
+	return s.CreateSheet(ctx, sheet)
+}
+
+func getSheetCommands(engine storepb.Engine, statement string) []*storepb.SheetCommand {
+	singleSQLs, err := base.SplitMultiSQL(engine, statement)
+	if err != nil {
+		if !strings.Contains(err.Error(), "not supported") {
+			slog.Warn("failed to split multi sql", "engine", engine.String(), "statement", statement)
+		}
+		return []*storepb.SheetCommand{
+			{
+				Start: 0,
+				End:   int32(len(statement)),
```

Comment:
Make sure we handle string index v.s. byte index and encoding correctly. But it looks correct to me.